### DISCUSSION TITLE
Loop once but timed

### DISCUSSION THREAD

**Discussion by chenziliang** [libuv/libuv#4317] [Q&A] [answered]
Hi,

May i ask what's the best practice to use UV_RUN_ONCE, but passing in a timeout value ? For now, I setup a timer every time before calling uv_run.

May I ask if there there any discussion regarding uv_run accepts a timeout parameter before ?

Thanks so much !

**Comment by bnoordhuis** [comment, marked as answer]
> I setup a timer every time before calling uv_run.

That's a fine approach.

> May I ask if there there any discussion regarding uv_run accepts a timeout parameter before ?

Libuv's v1.x branch (where releases are made from) is API- and ABI-stable. No backwards incompatible changes are accepted.

**Reply by chenziliang** [reply]
Thanks !
